lib/kubernetes-deploy/runner_task.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+require 'tempfile'
+
+require 'kubernetes-deploy/kubeclient_builder'
+require 'kubernetes-deploy/ui_helpers'
+require 'kubernetes-deploy/kubectl'
+
+module KubernetesDeploy
+  class RunnerTask
+    include KubernetesDeploy::KubeclientBuilder
+    include Kubectl
+
+    class FatalTaskRunError < FatalDeploymentError; end
+
+    def initialize(namespace:, context:, logger: KubernetesDeploy.logger)
+      @logger = logger
+      @namespace = namespace
+      @kubeclient = build_v1_kubeclient(context)
+      @context = context
+    end
+
+    def run(task_template:, command:, args:, env_vars: [])
+      validate_configuration(task_template, args)
+
+      raw_template = get_template(task_template)
+      rendered_template = build_pod_template(raw_template, command, args, env_vars)
+
+      validate_pod_spec(rendered_template)
+
+      KubernetesDeploy.logger.info("Starting task runner pod: '#{rendered_template.metadata.name}'")
+
+      @kubeclient.create_pod(rendered_template)
+    end
+
+    private
+
+    def validate_configuration(task_template, args)
+      errors = []
+
+      if task_template.blank?
+        errors << "Task template name can't be nil"
+      end
+
+      if args.blank?
+        errors << "Args can't be nil"
```

Comment:
Really the test should be that args and command can't be nil, at least one of the two should be set. However, in the interest of supporting our most common use case (shopify core) it seems like requiring `args` to be required is the most reasonable option.